src/components/flow/flow.tsx

```diff
@@ -119,44 +97,21 @@ export class Flow {
     return newPanelCount < oldPanelCount ? "retreating" : "advancing";
   };
 
-  handleMutationObserverChange = (): void => {
-    const { el, allowDescendantPanels } = this;
-
-    if (!allowDescendantPanels) {
-      return;
-    }
+  updateFlowProps = (): void => {
+    const { el, panels } = this;
 
     const newPanels: HTMLCalcitePanelElement[] = Array.from(
-      el.querySelectorAll("calcite-panel:not(calcite-panel calcite-panel)")
-    );
-
-    this.updateFlowProps(newPanels);
-  };
-
-  handleDefaultSlotChange = (event: Event): void => {
-    if (this.allowDescendantPanels) {
-      return;
-    }
-
-    const newPanels = (event.target as HTMLSlotElement)
-      .assignedElements({
-        flatten: true
-      })
-      .filter((el) => el?.matches("calcite-panel")) as HTMLCalcitePanelElement[];
+      el.querySelectorAll("calcite-panel")
+    ).filter((panel) => !panel.matches("calcite-panel calcite-panel")) as HTMLCalcitePanelElement[];
 
-    this.updateFlowProps(newPanels);
-  };
-
-  updateFlowProps = (newPanels: HTMLCalcitePanelElement[]): void => {
-    const { panels } = this;
     const oldPanelCount = panels.length;
     const newPanelCount = newPanels.length;
     const activePanel = newPanels[newPanelCount - 1];
     const previousPanel = newPanels[newPanelCount - 2];
 
     if (newPanelCount && activePanel) {
       newPanels.forEach((panelNode) => {
-        panelNode.showBackButton = newPanelCount > 1;
+        panelNode.showBackButton = panelNode === activePanel && newPanelCount > 1;
```

Comment:
Is this a separate fix? It seems unrelated to the panel query behavior.

Reply:
yeah ill separate it out.